core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala

```diff
@@ -1,37 +1,147 @@
 package dev.atedeg.ecscala.dsl
 
-import dev.atedeg.ecscala.util.types.ComponentTag
-import dev.atedeg.ecscala.{ Component, Entity, World }
-import dev.atedeg.ecscala.dsl.Words.*
+import dev.atedeg.ecscala.util.types.{ CListTag, ComponentTag }
+import dev.atedeg.ecscala.{ CList, Component, Deletable, Entity, System, View, World }
+import dev.atedeg.ecscala.dsl.Words.EntityWord
+import dev.atedeg.ecscala.DeltaTime
 
 /**
  * This trait provides a domain specific language (DSL) for expressing the Ecscala framework operation using an
- * english-like language.
+ * english-like language. Here's the things you can do:
+ *
+ * '''Create an entity in a world:'''
+ * {{{
+ * val world = World()
+ * val entity1 = world hasAn entity
+ * }}}
+ *
+ * '''Remove entities from a world:'''
+ * {{{
+ *   *  world - entity1
+ *   *  remove (entity1) from world
+ *   *  remove (entity1, entity2, entity3) from world
+ * }}}
+ *
+ * '''Create an entity in a world with a component:'''
+ * {{{
+ * val entity1 = world hasAn entity withComponent MyComponent()
+ * }}}
+ *
+ * '''Create an entity in a world with multiple components:'''
+ * {{{
+ * val entity1 = world hasAn entity withComponents {
+ *       MyComponent1() and MyComponent2() and MyComponent3()
+ * }
+ * }}}
+ *
+ * '''Add components to an entity:'''
+ * {{{
+ *   *  entity1 + MyComponent()
+ *   *  entity1 withComponent MyComponent()
+ *   *  entity1 withComponents { MyComponent1() and MyComponent2() }
+ * }}}
+ *
+ * '''Remove a components from an entity:'''
+ * {{{
+ *   *  remove MyComponent() from entity1
+ *   *  entity1 - MyComponent()
+ *   *  remove { MyComponent1() and MyComponent2 and MyComponent3 } from entity1
+ * }}}
+ *
+ * '''Add a system to a world:'''
+ * {{{
+ * world hasA system[MyComponent() &: CNil] { (_,_,_) => {}}
```

Comment:
Here you should not use parentheses, you are specifying a type, not passing an instance
```suggestion
 * world hasA system[MyComponent &: CNil] { (_,_,_) => {}}
```